internal/sidecarlogresults/sidecarlogresults_test.go

```diff
@@ -0,0 +1,148 @@
+			wantResults := []string{}
+			for _, result := range c.results {
+				createResult(t, dir, result.Name, result.Value)
+				resultNames = append(resultNames, result.Name)
+				encodedResult, err := json.Marshal(result)
+				if err != nil {
+					t.Error(err)
+				}
+				// encode adds a newline character at the end.
+				// We need to do the same before comparing
+				encodedResult = append(encodedResult, '\n')
+				wantResults = append(wantResults, string(encodedResult))
+			}
+			dir2 := t.TempDir()
+			go createRun(t, dir2, false, true)
+			got := new(bytes.Buffer)
+			err := LookForResults(got, dir2, dir, strings.Join(resultNames, ","))
+			if err != nil {
+				t.Fatalf("Did not expect any error but got: %v", err)
+			}
+			for _, res := range wantResults {
+				if !strings.Contains(got.String(), res) {
+					t.Errorf("Expected result %s to be in %s", res, got.String())
+				}
+			}
+		})
+	}
+}
+
+func TestLookForResults(t *testing.T) {
+	for _, c := range []struct {
+		desc         string
+		resultName   string
+		resultValue  string
+		createResult bool
+		stepError    bool
+	}{{
+		desc:         "good result",
+		resultName:   "foo",
+		resultValue:  "bar",
+		createResult: true,
+		stepError:    false,
+	}, {
+		desc:         "empty result",
+		resultName:   "foo",
+		resultValue:  "",
+		createResult: true,
+		stepError:    true,
+	}, {
+		desc:         "missing result",
+		resultName:   "missing",
+		resultValue:  "",
+		createResult: false,
+		stepError:    false,
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			dir := t.TempDir()
+			if c.createResult == true {
+				createResult(t, dir, c.resultName, c.resultValue)
+			}
+			dir2 := t.TempDir()
+			createRun(t, dir2, c.stepError, false)
+
+			want := []byte{}
+			if c.createResult == true {
+				// This is the expected result
+				result := SidecarLogResult{
+					Name:  c.resultName,
+					Value: c.resultValue,
+				}
+				encodedResult, err := json.Marshal(result)
+				if err != nil {
+					t.Error(err)
+				}
+				// encode adds a newline character at the end.
+				// We need to do the same before comparing
+				encodedResult = append(encodedResult, '\n')
+				want = encodedResult
+			}
+			got := new(bytes.Buffer)
+			err := LookForResults(got, dir2, dir, c.resultName)
+			if err != nil {
+				t.Fatalf("Did not expect any error but got: %v", err)
+			}
+			if d := cmp.Diff(want, []byte(got.String())); d != "" {
+				t.Errorf(diff.PrintWantGot(d))
+			}
+		})
+	}
+}
+
+func createResult(t *testing.T, dir string, resultName string, resultValue string) {
+	t.Helper()
+	resultFile := filepath.Join(dir, resultName)
+	err := os.WriteFile(resultFile, []byte(resultValue), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func createRun(t *testing.T, dir string, causeErr bool, sleep bool) {
+	t.Helper()
+	stepFile := filepath.Join(dir, "1")
+	_ = os.Mkdir(stepFile, 0755)
+	stepFile = filepath.Join(stepFile, "out")
+	if causeErr {
+		stepFile += ".err"
+	}
+	if sleep {
```

Comment:
This is always false. Remove?

(since this is setting up the results this won't actually cause variation in the goroutine behavior later)

Reply:
sure. Sounds good.